samples/net/openthread/coap_client/prj.conf

```diff
@@ -0,0 +1,88 @@
+# Generic networking options
+CONFIG_NETWORKING=y
+CONFIG_NET_UDP=y
+CONFIG_NET_IPV6=y
+CONFIG_NET_SOCKETS=y
+CONFIG_NET_SOCKETS_POSIX_NAMES=y
+CONFIG_NET_SOCKETS_POLL_MAX=4
+CONFIG_NET_CONNECTION_MANAGER=y
+
+# Enable CoAP protocol
+CONFIG_COAP=y
+CONFIG_HEAP_MEM_POOL_SIZE=2048
+
+# Kernel options
+CONFIG_MAIN_STACK_SIZE=2048
+CONFIG_ENTROPY_GENERATOR=y
+CONFIG_TEST_RANDOM_GENERATOR=n
+CONFIG_INIT_STACKS=y
+
+# Logging
+CONFIG_NET_LOG=y
+CONFIG_LOG=y
+CONFIG_NET_STATISTICS=y
+CONFIG_PRINTK=y
+
+# Network buffers
+CONFIG_NET_PKT_RX_COUNT=16
+CONFIG_NET_PKT_TX_COUNT=16
+CONFIG_NET_BUF_RX_COUNT=80
+CONFIG_NET_BUF_TX_COUNT=80
+CONFIG_NET_CONTEXT_NET_PKT_POOL=y
+
+# IP address options
+CONFIG_NET_IF_UNICAST_IPV6_ADDR_COUNT=3
+CONFIG_NET_IF_MCAST_IPV6_ADDR_COUNT=4
+CONFIG_NET_MAX_CONTEXTS=10
+
+# Network shell
+CONFIG_NET_SHELL=y
+
+CONFIG_CPLUSPLUS=y
+
+CONFIG_BT=n
+
+CONFIG_NET_CONFIG_SETTINGS=y
+CONFIG_NET_TCP=n
+CONFIG_NET_IPV4=n
+CONFIG_NET_IPV6_NBR_CACHE=n
+CONFIG_NET_IPV6_MLD=n
+CONFIG_NET_CONFIG_NEED_IPV6=y
+CONFIG_NET_CONFIG_NEED_IPV4=n
+CONFIG_NET_CONFIG_MY_IPV4_ADDR=""
+CONFIG_NET_CONFIG_PEER_IPV4_ADDR=""
+
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=2048
+CONFIG_NET_TX_STACK_SIZE=2048
+CONFIG_NET_RX_STACK_SIZE=5120
+CONFIG_SHELL_STACK_SIZE=3072
+
+CONFIG_NET_L2_OPENTHREAD=y
+
+CONFIG_OPENTHREAD_DEBUG=y
+CONFIG_OPENTHREAD_L2_DEBUG=y
+CONFIG_OPENTHREAD_L2_LOG_LEVEL_INF=y
+
+CONFIG_OPENTHREAD_CHANNEL=21
+
+CONFIG_NET_CONFIG_MY_IPV6_ADDR="fdde:ad00:beef::1"
```

Comment:
Thank Robert, having some kind of service discovery is very important. Łukasz will need to add MTD device example in the short term, so we may check all the implications of adding the address statically.

I'm not sure if we need any static address allocation for CoAP examples as they are defined right now, right @lmaciejonczyk ?

Reply:
Like @rlubos said, without it:

> Zephyr networking stack can hang up for ~30s during initialization.

Reply:
Just to elaborate on that, the delay is specified by this config:
https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/net/lib/config/Kconfig#L19

So an alternative, for now, would be to make this delay smaller, just to allow the OT stack to join the network. Eventually though, OT should throw some kind of network ready event to the stack, so that the application can start just after the connection is established.

Reply:
Set: `CONFIG_NET_CONFIG_INIT_TIMEOUT=3` Looks to be enough.